paymentsheet/src/main/java/com/stripe/android/paymentsheet/AddPaymentMethodsAdapter.kt

```diff
@@ -74,42 +74,120 @@ internal class AddPaymentMethodsAdapter(
+            val minItemWidth = 100 * screenDensity + (2 * CARD_HORIZONTAL_PADDING)
+
+            // if all items fit at min width, then span them across the sheet evenly filling it.
+            // otherwise the number of items visible should be a multiple of .5
+            val viewWidth =
+                if (minItemWidth * numberOfPaymentMethods < targetWidth) {
+                    targetWidth / numberOfPaymentMethods
+                } else {
+                    // numVisibleItems is incremented in steps of 0.5 items
+                    // (1, 1.5, 2, 2.5, 3, ...)
+                    val numVisibleItems = (targetWidth * 2 / minItemWidth).toInt() / 2f
+                    targetWidth / numVisibleItems
+                }
+
+            return (viewWidth.toInt() / screenDensity).dp
+        }
+
+        internal const val ADD_PM_DEFAULT_PADDING = 12.0f
+        internal const val CARD_HORIZONTAL_PADDING = 6.0f
+    }
+}
+
+@Composable
+internal fun PaymentMethodUI(
+    viewWidth: Dp,
+    iconRes: Int,
+    title: String,
+    isSelected: Boolean,
+    isEnabled: Boolean,
+    onItemSelectedListener: () -> Unit
+) {
+    val strokeColor = colorResource(
+        if (isSelected) {
+            R.color.stripe_paymentsheet_add_pm_card_selected_stroke
+        } else {
+            R.color.stripe_paymentsheet_add_pm_card_unselected_stroke
         }
+    )
 
-        fun bind(paymentMethod: SupportedPaymentMethod) {
-            binding.icon.setImageResource(paymentMethod.iconResource)
-            binding.title.text = itemView.resources.getString(paymentMethod.displayNameResource)
+    val cardBackgroundColor = if (isEnabled) {
+        colorResource(R.color.stripe_paymentsheet_elements_background_default)
+    } else {
+        colorResource(R.color.stripe_paymentsheet_elements_background_disabled)
+    }
+
+    Card(
+        border = BorderStroke(if (isSelected) 2.dp else 1.5.dp, strokeColor),
+        shape = RoundedCornerShape(6.dp),
+        elevation = if (isSelected) 1.5.dp else 0.dp,
+        backgroundColor = cardBackgroundColor,
+        modifier = Modifier
+            .height(60.dp)
+            .width(viewWidth)
+            .padding(horizontal = CARD_HORIZONTAL_PADDING.dp)
+            .selectable(
+                selected = isSelected,
+                enabled = isEnabled,
+                onClick = {
+                    onItemSelectedListener()
+                }
+            )
+    ) {
+        Column {
+            Image(
+                painter = painterResource(iconRes),
+                contentDescription = "",
+                modifier = Modifier
+                    .padding(top = ADD_PM_DEFAULT_PADDING.dp, start = ADD_PM_DEFAULT_PADDING.dp)
+            )
+            LpmSelectorText(
+                text = title,
+                isEnabled = isEnabled,
+                modifier = Modifier.padding(top = 6.dp, start = ADD_PM_DEFAULT_PADDING.dp)
```

Comment:
Should it really be only top and start? Don't we want to also have a padding at the end?

Reply:
Currently in Master it looks like this:
![Screen Shot 2022-02-25 at 9 59 14 AM](https://user-images.githubusercontent.com/89166418/155766782-ad55daf8-7879-47a3-b218-df4cfd3a926a.png)

I'm going to sync with JJ and ask what to do in this situation, but not using end padding makes it look like my screenshot above which I think is better.

Reply:
Maybe ellipsize for now? 
`maxLines = 1, overflow = TextOverflow.Ellipsis`

Reply:
That's better. 
![Screen Shot 2022-02-25 at 11 09 54 AM](https://user-images.githubusercontent.com/89166418/155773839-e5283961-bb7c-48df-8b86-287d3fa1325e.png)